File: frontend/types/task.ts
export interface Task {
  TaskID: number;
  Title: string;
  Description?: string;
  AssignedTo: number;
  Priority: string;
  Status: string;
  DueDate: string;
}
export type TaskCreate = Omit<Task, "TaskID">;
export type TaskUpdate = Partial<TaskCreate>;

File: backend/app/schemas/task.py
from pydantic import BaseModel
from datetime import datetime


class TaskCreate(BaseModel):
    Title: str
    Description: str
    AssignedTo: int
    Priority: str
    Status: str
    DueDate: datetime


class TaskUpdate(BaseModel):
    Title: str
    Description: str
    AssignedTo: int
    Priority: str
    Status: str
    DueDate: datetime